test/e2e/framework/node/wait.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/util/system"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+	testutils "k8s.io/kubernetes/test/utils"
+)
+
+var requiredPerNodePods = []*regexp.Regexp{
+	regexp.MustCompile(".*kube-proxy.*"),
+	regexp.MustCompile(".*fluentd-elasticsearch.*"),
+	regexp.MustCompile(".*node-problem-detector.*"),
+}
+
+// WaitForReady waits up to timeout for cluster to has desired size and
+// there is no not-ready nodes in it. By cluster size we mean number of Nodes
+// excluding Master Node.
+func WaitForReady(c clientset.Interface, size int, timeout time.Duration) error {
```

Comment:
Why change the signature of this method from `e2enode.WaitForReadyNodes` to `e2enode.WaitForReady`?
1WaitForReady` could wait for any general condition while `WaitForReadyNodes` makes it sound more explicit that the condition is a certain number of nodes

Reply:
As per Tim's suggestion, we have a `e2enode` package, so the signature should remove the redundant `node(s)` in naming. `e2enode.WaitForReady` has a `e2enode` prefix, so people know that this is a node related function

what do you think?

Reply:
Makes sense removing the redundant naming but for this one, if i was looking at godos (as an example), `e2enode.WaitForReadyNodes` sounds like it waits for some number of nodes, while `e2enode.WaitForReady` sounds like i could pass some "condition" object and check if a certain node meets that condition.

Reply:
👍 make more sense